### ISSUE TITLE
sort-vars do not support const & let

### ISSUE THREAD

**Issue by cmalard** [eslint/eslint#8765]
**Tell us about your environment**

* **ESLint Version:** 4.0.0
* **Node Version:** 8.1.1
* **npm Version:** 5.0.3

**What parser (default, Babel-ESLint, etc.) are you using?** Babel-ESLint

**Please show your full configuration:**

```json
{
  "env": {
    "es6": true,
  },
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
  },
  "rules": {
    "sort-vars": ["error", { "ignoreCase": true }],
  },
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
  const b = 20;
  let d = 40;
  const a = 10;
  let c = 30;
```

**What did you expect to happen?**
sort-vars should trigger an error.
The correct code should be this :
```js
  const a = 10;
  const b = 20;
  let c = 30;
  let d = 40;
```

**What actually happened? Please include the actual, raw output from ESLint.**
nothing

**Comment by cmalard** [comment]
well, I went too fast, sort-vars only works for the same block `var a, b, c;`
